Lib/pathlib.py

```diff
@@ -827,7 +831,14 @@ def absolute(self):
         """
         if self.is_absolute():
             return self
-        return self._from_parts([os.getcwd()] + self._parts)
+        elif self._drv and _getfullpathname:
+            try:
+                cwd = _getfullpathname(self._drv)
+            except (ValueError, OSError):
```

Comment:
I think the only `ValueError` exceptions raised by `_getfullpathname()` are either from a decoding error of a `bytes` path or from embedded null characters in the path. I think the only `OSError` that can occur is if the path is an empty string or just spaces. None of these should apply to `self._drv` in this case. If they do occur for some reason, do we really want to ignore an invalid drive here?